src/Workspaces/Remote/ServiceHub/Shared/ServerDirectStream.cs

```diff
@@ -9,93 +9,98 @@
 
 namespace Microsoft.CodeAnalysis.Remote
 {
-    /// <summary>
-    /// Direct stream between service hub server and client to pass around big chunk of data
-    /// </summary>
-    internal class ServerDirectStream : Stream
+    internal static partial class Extensions
     {
-        // 128KB buffer size
-        private const int BUFFERSIZE = 128 * 1024;
+        /// <summary>
+        /// Direct stream between service hub server and client to pass around big chunk of data.
+        /// 
+        /// This stream should be only consumed through JsonRpc.InvokeAsync
+        /// </summary>
+        private class ServerDirectStream : Stream
```

Comment:
Why do we need ClientDirectStream and ServerDirectStream?  Could we share them?  It seems unfortunate that you have two classes that just subclass stream, just to forward to a buffered stream that wraps a pipe.  can we just have one type for this, and it can be used in either direction (maybe through a boolean passed to the parameter stating which direction you're going in?)

Then we could have helper static methods to actually create it which would hide the bool.  Like "CreateStreamToServer" "CreateStreamToClient" or something like that.